### PR TITLE
anagram: Adding test template

### LINKED ISSUE #1926: anagram: add test template
## Task
Create a test template file for `anagram`

### Details
1. If this is your first time contributing to this repository, welcome!
    - Contributing to a repository on GitHub requires a [fork](https://help.github.com/en/articles/fork-a-repo)
1. Create a branch on your local repository called `anagram-add-test-template`
    - Run: `git checkout -b anagram-add-test-template`
1. Clone problem-specifications repository; this is used to populate the template
    - Run: `git clone https://github.com/exercism/problem-specifications ../problem-specifications`
1. Create `exercises/anagram/.meta/template.j2`
    - See [Generator Documentation] for more information
1. Generate test file from new template
    - Run: `bin/generate_tests.py anagram`

At this point, if the generator fails, or you are otherwise stuck,
please feel free to reach out! It is perfectly acceptable to submit
an incomplete PR in order to get feedback or assistance.

### Resources
- [Contributing Guide]
- [Generator Documentation]
- [Closing issues using keywords]

[Contributing Guide]: https://github.com/exercism/python/blob/master/docs/CONTRIBUTING.md
[Generator Documentation]: https://github.com/exercism/python/blob/master/docs/GENERATOR.md
[Closing issues using keywords]: https://help.github.com/en/articles/closing-issues-using-keywords

### PR DESCRIPTION
**Pull request by irshadshalu**
Added `exercies/anagram/.meta/template.j2` and corresponding tests.

Resolves #1926

### CODE REVIEW

**Comment by irshadshalu** [comment]
@yawpitch , Made the required changes.

File: exercises/anagram/.meta/template.j2 [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{%- import "generator_macros.j2" as macros with context -%}
+
+# Python 2/3 compatibility
+if not hasattr(unittest.TestCase, 'assertCountEqual'):
+    unittest.TestCase.assertCountEqual = unittest.TestCase.assertItemsEqual
```
Comment (cmccandless) [inline exercises/anagram/.meta/template.j2]: The `import unittest` line is inside the header; this section will need to be moved after that.

File: exercises/anagram/anagram_test.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-
 import unittest
 
 from anagram import find_anagrams
 
 # Tests adapted from `problem-specifications//canonical-data.json` @ v1.5.0
 
+# Python 2/3 compatibility
+if not hasattr(unittest.TestCase, "assertCountEqual"):
+    unittest.TestCase.assertCountEqual = unittest.TestCase.assertItemsEqual
```
Comment (cmccandless) [inline exercises/anagram/anagram_test.py]: I'm confused: why is this needed? `assertCountEqual` is not referenced in this file.

Comment (yawpitch) [reply exercises/anagram/anagram_test.py]: It's mentioned in the non-template test file, in the test for two and three anagrams, which happen to be the only tests that return multiple anagrams.

I don't believe we've got any requirement for the output to be sorted and _maybe_ someone wants to parallelize?

Comment (irshadshalu) [reply exercises/anagram/anagram_test.py]: > I'm confused: why is this needed? assertCountEqual is not referenced in this file.
I was confused about the need of this portion of code as well which is I removed it in the first place. 

> I don't believe we've got any requirement for the output to be sorted and maybe someone wants to parallelize?
In that case, is it better to replace `assertEqual` with `assertItemsEqual` in the template and this compatibility code can be removed as well?

Comment (yawpitch) [reply exercises/anagram/anagram_test.py]: Well no, we’d need to keep the compatibility code and convert assertEqual to assertCountEqual, since assertItemsEqual was removed in 3.
